From 93db71781969830f7c42714e1e15e5ddfd56daf7 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 30 Sep 2013 10:29:21 -0700
Subject: [PATCH 318/974] MIPS: OCTEON: Fix typos in octeon-irq.c

The CIB block handlers were fubared.

Signed-off-by: David Daney <david.daney@cavium.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/cavium-octeon/octeon-irq.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/octeon-irq.c b/arch/mips/cavium-octeon/octeon-irq.c
index f8f7984..c9fbea8 100644
--- a/arch/mips/cavium-octeon/octeon-irq.c
+++ b/arch/mips/cavium-octeon/octeon-irq.c
@@ -2581,7 +2581,7 @@ static void octeon_irq_cib_enable(struct irq_data *data)
 
 	raw_spin_lock_irqsave(&host_data->lock, flags);
 	en = cvmx_read_csr(host_data->en_reg);
-	en |= 1ull < cd->bit;
+	en |= 1ull << cd->bit;
 	cvmx_write_csr(host_data->en_reg, en);
 	raw_spin_unlock_irqrestore(&host_data->lock, flags);
 }
@@ -2595,13 +2595,13 @@ static void octeon_irq_cib_disable(struct irq_data *data)
 
 	raw_spin_lock_irqsave(&host_data->lock, flags);
 	en = cvmx_read_csr(host_data->en_reg);
-	en &= ~(1ull < cd->bit);
+	en &= ~(1ull << cd->bit);
 	cvmx_write_csr(host_data->en_reg, en);
 	raw_spin_unlock_irqrestore(&host_data->lock, flags);
 }
 
 static struct irq_chip octeon_irq_chip_cib = {
-	.name = "CIU3",
+	.name = "CIB",
 	.irq_enable = octeon_irq_cib_enable,
 	.irq_disable = octeon_irq_cib_disable,
 	.irq_mask = octeon_irq_cib_disable,
-- 
2.6.2

